.github/scripts/update-kotlin-files.py

```
@@ -0,0 +1,152 @@
+            if not file_strings:
+                return False
+            
+            print(f"📝 Updating {file_path} with {len(file_strings)} strings")
+            
+            # Update each string reference
+            for string_info in sorted(file_strings, key=lambda x: x['line_number'], reverse=True):
+                old_string = string_info['original_string']
+                new_key = string_info['semantic_key']
+                
+                # Replace string literal with resource reference
+                old_pattern = re.escape(f'"{old_string}"')
```

Ruff (0.14.14)

[error] Local variable `old_pattern` is assigned to but never used

Remove assignment to unused variable `old_pattern`

(F841)